compiler/src/typed/translprim.re

```diff
@@ -3,7 +3,13 @@ open Ast_helper;
 open Typedtree;
 open Parsetree;
 
+type wi32_constant =
+  | HeapBase
+  | HeapStart
+  | HeapTypeMetadataPointer;
+
 type primitive =
+  | PrimitiveWasmI32(wi32_constant)
```

Comment:
Could we instead call this `PrimitiveConst` and call `wi32_constant` `prim_constant`? I definitely see us dropping in other types as well! (Want to reserve Primitive0 for zero-arg functions.)

Reply:
So, the reason why I called it that was _because_ I expect us to drop in other types 🙂. I would imagine that we could have `PrimitiveWasmI64`, etc...I think that makes sense, no?

Reply:
Hmm, I don't really see this as being any different than the other ones (Primitive1, etc) and we wouldn't have special versions for different typed functions. I think it might get annoying if we started to add a bunch of these for all of the different types we might have.

Reply:
I saw them as different because we are not wrapping them with lambdas.

Reply:
A value's a value!